### PR TITLE
New Dapr Components Metadata Format + Tutorial on Referencing Secrets in Dapr Components

### PR DESCRIPTION
**Pull request by SoTrx**
## Description
Hey,

This PR updates all Dapr components to the new metadata format and adds a guide on how to use secret references in Dapr components.

## Issue reference
radius-project/radius#7704

### CODE REVIEW

File: docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+type: docs
+title: "How-To: Reference secrets in Dapr components"
+linkTitle: "Reference secrets in components"
+description: "Learn how to manage secrets in Dapr components"
+weight: 300
+categories: "How-To"
+tags: ["Dapr"]
+---
+
+This how-to guide will provide an overview of how to:
```
Comment (willtsai) [inline docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md]:
```suggestion
This guide will provide an overview of how to:
```

**Review by ytimocin — APPROVED** [review]

File: docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+- [rad CLI]({{< ref "installation#step-1-install-the-rad-cli" >}})
+- [Bicep VSCode extension]({{< ref "installation#step-2-install-the-vs-code-extension" >}})
+- [Radius environment]({{< ref "installation#step-3-initialize-radius" >}})
+- [Radius local-dev Recipes]({{< ref howto-dev-recipes >}})
+- [Dapr installed on your Kubernetes cluster](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-deploy/)
+
+## Step 1: Start with a container and a Dapr sidecar
+
+Begin by creating a file named app.bicep, which defines a container with a Dapr state store. If you need a detailed explanation on how to do so, refer to the [Add a Dapr building block]({{< ref how-to-dapr-building-block >}}) tutorial.
```
Comment (willtsai) [inline docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md]:
```suggestion
Begin by creating a file named `app.bicep`, which defines a container with a Dapr state store. If you need a detailed explanation on how to do so, refer to the [Add a Dapr building block]({{< ref how-to-dapr-building-block >}}) tutorial.
```

File: docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+## Prerequisites
+
+- [rad CLI]({{< ref "installation#step-1-install-the-rad-cli" >}})
+- [Bicep VSCode extension]({{< ref "installation#step-2-install-the-vs-code-extension" >}})
+- [Radius environment]({{< ref "installation#step-3-initialize-radius" >}})
+- [Radius local-dev Recipes]({{< ref howto-dev-recipes >}})
+- [Dapr installed on your Kubernetes cluster](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-deploy/)
+
+## Step 1: Start with a container and a Dapr sidecar
```
Comment (willtsai) [inline docs/content/guides/author-apps/dapr/how-to-dapr-secrets/index.md]:
```suggestion
## Step 1: Create a container and a Dapr sidecar
```

**Review by willtsai — APPROVED** [review]
some minor edits which i'll commit directly, but otherwise looks great! thank you for the contribution @SoTrx

**Pull request merged**